### DISCUSSION TITLE
New water meter can't be read reliably

### DISCUSSION THREAD

**Discussion by garciaargos** [jomjol/AI-on-the-edge-device#3109] [General]
### Issue

I'm not sure how to classify this, so I chose "blank form". Today we got our water meter replaced, with one of those that can be made "smart" by adding an impulse meter. And it's so poorly designed that I can't even see the 3rd analogue digit properly by looking at it. In the image below you can see what I mean.

Obviously, the AI-on-the-edge has a hard time with it. It seems it even gets the image mirrored, so I turned on anti-CW for that digit. But I still get a lot of errors due to this digit. 

Perhaps there is a way to make the analogue digits less relevant if the recognition is not so good? Or can we ignore some digits?

In the end, the whole image is a big mess in the analogue digits area, because the way the transparent part is made deforms the image, so I'm not sure any image recognition is going to work with this.

Time to spend money on the impulse meter? Is this usable for AI-on-the-edge? Or do I complain to the Wasserwerk because I can't even read it properly with my bare eyes?

![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/7721696/62d6ec50-b03e-42a9-83f6-bb67c4dc7572)

**Comment by caco3** [comment]
Throw the last 2 pointers away, they can't be used with the AIOTED.even the 3th last pointer might be an issue as it is partially covered. But the most right pointer and the digits are very well, they should cause no issues. Use the [Extendedn Resolution](https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#PostProcessing-NUMBER.ExtendedResolution) parameter to increase the resolution of the pointer.
You image is a bit blurry, try to work on this.
and the digit ROIs seem too large, check the documentation to set them properly.

**Reply by garciaargos** [reply]
Thanks for the reply. I would like to keep the last few digits to detect leaks overnight (which has happened before in this old house, last time start of this spring). I was playing around with the digital ROIs sizes because it read the first 0 as a 3 and a 4 a few times. Extended resolution is already on, which is maybe helping it a bit. This system had been running for a bit over a year!

If by blurry you mean the deformation of the "x0,0001" text and things like that, that seems to be caused by the... water inside the meter!

Nevertheless, with the water inside, some droplets randomly appeared and there's even distortion to some of the digital digits. The company that installed it said it's normal and we are stuck with this awful design. So far this meter seems very much incompatible with AI-on-the-edge :(

But I'm still going to try further, by removing one of the pieces of the holder to make the camera closer to the meter and deleting the larger digits (100, 1000, and 10000) since they are irrelevant now (and most likely in the next 6 years) and the water drops just introduce that error. 

I just realised after looking at a picture I took with my phone that the 3rd analogue digit is really the numbers rotating and not an arrow (see the picture I took with my phone below). This probably changes things a bit, as with the closer image, one can more or less read the numbers that are on top (where the marker is). I don't see any meters like that in the documentation, would it help sending some images to train a model for this kind of digit? It would then need to add a config setting for the digits.

![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/7721696/5b27b0ac-a9c8-43d7-9fdf-f4673a65e0a6)

**Reply by caco3** [reply]
Dooh, I would have expected that water in the display part is a bad sign, but according to https://www.haustechnikdialog.de/Forum/t/53390/Wasser-in-der-Wasseruhr- this seems to be common and ok :(

The moving air bubbles clearly will be an issue. That could only be fixed by hardware, eg. maybe by tilting the meter a bit.

You could try to create your own model. If you train the 2nd pointer from the right, that might work.
But the 2nd one from the left (the black one) will most likely not work. The reason is that the analog models try to calculate the angle of the pointer, but there is no pointer on that one.
Also the firmware does not support to use it as a digit. You would need to create a 2nd number with just that digit and then do a post processing outside the device to combine the information of both numbers. IMO its tricky and would not be reliable.

**Reply by Slider0007** [reply]
@garciaargos:

I have the same meter in place. In the beginning I also was kind of disappointed by this strange hardware design...
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/115730895/71ea6647-6ede-494c-9d44-660932c8c5a8)

- If using only the higher two analogs no model re-training is needed. I'm facing no issues at all with default models (using a class 100 model).
- Like already stated by caco3 the black counter type is not supported by firmware
- Usage of extended resolution is recommended
- Don't worry the water will flood the meter completly after a while (it take some time, few weeks) -> no bubbles anymore
- To detect small leackage you could use a second sequence only containing the lowest analog counter and do post logic processing offline

**Reply by garciaargos** [reply]
@Slider0007  @caco3  I appreciate the time and the replies. Thanks for the tip about the water eventually filling everything up, that should make the image less crappy. 

I've rotated things physically so I don't need to adjust the angle in the image post-processing, that way everything is a bit clearer. 

I'll eventually drop the last two digits as you both have suggested, but if I get the time, I might try to get that black digit implemented. Chances are I won't get the time, but dreaming is free :) 

Honestly the lack of precision (or correct information for that matter) in the 3rd analogue digit (litres) isn't as critical to detecting the small leaks I've had to chase, it's the last one (100 mL) that comes in handy when checking what happened overnight. Since I send the data to an Influx database that I can later check on a Grafana dashboard, if it wildly guesses the L and shows me the 100mL just fine, I'm good. In case of suspected leak I do check the pictures if the data isn't clear anyway.

For monthly and yearly readings, the first two analogue digits are certainly enough.

**Comment by friedpa** [comment]
Maybe a suggestion you might follow. I bought a water meter by myself (they are cheap) and installed it after the "offical" meter. I bought this one:

![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/42890695/e20a878d-2281-4e16-9ea8-11844e519512)

Perfect readings, also kalibrated...

**Reply by garciaargos** [reply]
Believe it or not, I thought about this myself. I measured that I have enough space to fit one, but I think I should also replace the filter and the pressure regulator that were installed who knows how long ago :)
